### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by BenBlaise**
Rewrote a few parts of the README, added some sections to give more context on the project and to be friendlier towards newcomers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,82 @@
...
+
+The processing functions defined in the user's nodes accept a packet to be processed as a parameter. After they are done processing a packet, packets can be sent onto the next channel (one that the user node has knowledge of) in the pipeline by acquiring a packet to be filled from that channel and filling it with the processed data. The filled packet is automatically queued for processing in the next channel upon the destruction of the stub beloning to the filled packet. The space in memory belonging to the processed packet is freed up for reuse automatically upon the destruction of the stub belonging to the packet.
```
Comment (gerazo) [inline README.md:33]: "belonging to the filled packet" instead of beloning

File: README.md [hunk 2]
```diff
@@ -1,65 +1,82 @@
...
- 3. What you need for your project:
-   - `include` folder
-   - `build/libpipert.so` dynamic library _OR_
-   - `build/libpipert_static.a` static library
- 4. Find the documentation under `build/docs/index.html`
-
+1. Download the repo using `git clone https://github.com/gerazo/pipert.git`
+2. Run `./build.sh` to build the static and dynamic library in release mode
+3. What you need for your project:
+    - an `include` folder
```
Comment (gerazo) [inline README.md:52]: Here, the developer needs our include folder, not a random one. This was hazy in the original as well.

**Review by gerazo — COMMENTED** [review]
I think you could make it much more approachable. Thanks. I've found only 2 small problems.

File: README.md (hunk 1 above)
Comment (BenBlaise) [reply README.md]: That little typo managed to slip through, thanks.

File: README.md (hunk 2 above)
Comment (BenBlaise) [reply README.md]: Yeah, I wasn't sure about that, and I only suspected it vaguely. Thanks for clarifying.

**Review by gerazo — APPROVED** [review]

**Pull request merged**